flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/CustomDistributionITCase.java

```diff
@@ -94,12 +104,10 @@ public void testRangeWithDistribution2() throws Exception{
 			@Override
 			public void mapPartition(Iterable<Tuple3<Integer, Integer, String>> values, Collector<Boolean> out) throws Exception {
 				int partitionIndex = getRuntimeContext().getIndexOfThisSubtask();
-
 				for (Tuple3<Integer, Integer, String> s : values) {
-					if (s.f0 <= partitionIndex * (partitionIndex + 1) / 2 ||
-							s.f0 > (partitionIndex + 1) * (partitionIndex + 2) / 2 ||
-							s.f1 - 1 != partitionIndex) {
-						fail("Record was not correctly partitioned: " + s.toString());
+					
+					if ((s.f0 > partitionIndex + 1) || ((s.f0 == partitionIndex + 1) && (s.f1 > dist.rightBoundary[partitionIndex]))) {
```

Comment:
The test checks only one bucket boundary. We need to check 
- the upper boundary if `partitionIndex == 0`
- lower and upper boundaries if `0 < `partitionIndex` < 4`
- the lower boundary if `partitionIndex == 4.